engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -784,10 +784,15 @@ public void monitorHeartbeat(boolean waitForRecovery)
             containerStopRequests.put(c.getExternalId(), c.getExternalId());
           }
         } else {
-          if (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis) {
+          if (sca.lastHeartbeatMillis != -1 && (currentTms - sca.lastHeartbeatMillis > this.vars.heartbeatTimeoutMillis)) {
```

Comment:
Thank you for the review comment. What do you think about the repetitive log message? According to me we should only have one log message (like we are having after these code changes).